google-cloud-pub-sub-grpc/src/main/scala/akka/stream/alpakka/googlecloud/pubsub/grpc/javadsl/GooglePubSub.scala

```diff
@@ -88,13 +88,27 @@ object GooglePubSub {
 
         Source
           .tick(Duration.ZERO, pollInterval, request)
-          .mapAsync(1, client.pull)
-          .mapConcat(japiFunction(_.getReceivedMessagesList))
-          .mapMaterializedValue(japiFunction(cancellable.complete))
-          .mapMaterializedValue(japiFunction(_ => cancellable))
+          .mapAsync(1, client.pull(_))
+          .mapConcat(_.getReceivedMessagesList)
+          .mapMaterializedValue(cancellable.complete(_))
+          .mapMaterializedValue(_ => cancellable)
       }
-      .mapMaterializedValue(japiFunction(flattenCs))
-      .mapMaterializedValue(japiFunction(_.toCompletableFuture))
+      .mapMaterializedValue(flattenCs(_))
+      .mapMaterializedValue(_.toCompletableFuture)
+
+  /**
+   * Create a flow that accepts consumed message acknowledgements.
+   */
+  def acknowledgeFlow(): Flow[AcknowledgeRequest, AcknowledgeRequest, NotUsed] =
+    Flow
+      .setup { (mat, attr) =>
+        Flow
+          .create[AcknowledgeRequest]
+          .mapAsyncUnordered(1,
```

Comment:
[minor] I suppose this could just be `mapAsync` if parallelism always equals 1.

Reply:
Yes, `mapAsync` with parallelism 1 falls back to the `mapAsyncUnordered` implementation anyway.